flow/Platform.cpp

```diff
@@ -1666,6 +1666,27 @@ void atomicReplace( std::string const& path, std::string const& content, bool te
 		f = textmode ? fopen( tempfilename.c_str(), "wt" ) : fopen(tempfilename.c_str(), "wb");
 		if(!f)
 			throw io_error();
+	#ifdef _WIN32
```

Comment:
Should we do Windows too?

Reply:
I think Windows does this right by default. But we should at least test it.

Reply:
Looking at the documentation, it looks like ReplaceFile is good by itself - ie, it preserves all the attributes of the file being replaced. here is the relevant link and the text that is relevant for this discussion:

Link: 
here https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-replacefilea

> The ReplaceFile function combines several steps within a single function. An application can call ReplaceFile instead of calling separate functions to save the data to a new file, rename the original file using a temporary name, rename the new file to have the same name as the original file, and delete the original file. Another advantage is that ReplaceFile not only copies the new file data, but also preserves the following attributes of the original file:
> Creation time
> Short file name
> Object identifier
> DACLs
> Security resource attributes
> Encryption
> Compression
> Named streams not already in the replacement file
> For example, if the replacement file is encrypted, but the replaced file is not encrypted, the resulting file is not encrypted.
> Windows 7, Windows Server 2008 R2, Windows Server 2008, Windows Vista, Windows Server 2003 and Windows XP:  Security resource attributes (ATTRIBUTE_SECURITY_INFORMATION) for the original file are not preserved until Windows 8 and Windows Server 2012.

Reply:
Ok, cool. Can we throw in a comment to that effect so that this is clear to future readers?